src/TextInput2/TextInput2.tsx

```diff
@@ -83,7 +89,16 @@ const TextInput2: React.FC<Props> = ({
 
   useEffect(() => {
     // don't show error if nothing has happened yet
-    if (!isFocused && errorMessage === null) {
+    if (!showErrorOnBlurOnly && !isFocused && errorMessage === null) {
```

Comment:
This change should be backwards compatible, since existing implementations of `TextInput2` won't provide the `showErrorOnBlurOnly` prop- hence, `! showErrorOnBlurOnly` will evaluate to `true` for those component implementations.